qcodes/instrument/base_representation.py

```diff
@@ -0,0 +1,104 @@
+""" Representation methods for instruments """
+import logging
+import uuid
+from collections import namedtuple
+from functools import partial
+from html import escape
+from typing import Any, Dict, Optional
+
+from xarray.core.formatting_html import _icon, _mapping_section, _obj_repr
+from xarray.core.options import DISPLAY_EXPAND_ATTRS
+
+import qcodes
+
+Entry = namedtuple("Entry", ["name", "short", "long"])
+
+
+def create_entry(ed: Entry, preview: Optional[str] = None) -> str:
+    name = ed.name
+    preview = ed.short
+    long = ed.long
+
+    data_id = "data-" + str(uuid.uuid4())
+    data_icon = _icon("icon-database")
+    data_repr = f"{long}"
+    cssclass_idx = " class='xr-has-index'"
+
+    logging.debug(f"create_entry: {name} {preview}")
+
+    disabled = "" if long is not None else "disabled"
+
+    return (
+        f"<div class='xr-var-name'><span{cssclass_idx}>{name}</span></div>"
+        f"<div class='xr-var-preview xr-preview'>{preview}</div>"
+        f"<input id='{data_id}' class='xr-var-data-in' type='checkbox' {disabled}>"
+        f"<label for='{data_id}' title='Show/Hide data repr'>"
+        f"{data_icon}</label>"
+        f"<div class='xr-var-data'>{data_repr}</div>"
+    )
+
+
+def create_entries(variables: Dict[str, Entry]) -> str:
```

Comment:
could you please make all functions/variables apart from ``instrument_repr_html`` private (so that it's clear that they can change without notice)?